src/Symfony/Bridge/Twig/Extension/DumpExtension.php

```diff
@@ -67,11 +70,14 @@ public function dump(\Twig_Environment $env, $context)
         }
 
         $dump = fopen('php://memory', 'r+b');
-        $dumper = new HtmlDumper($dump);
+        $prevOutput = $this->dumper->setOutput($dump);
```

Comment:
@nicolas-grekas Using the 2nd arg doesn't work as it resets the header sent flag which causes tests to fail (due to recursion I'm assuming).

Even now isn't ideal with the header being dumped once per twig call. Thoughts? Maybe we set the header to empty string after the first dump call?